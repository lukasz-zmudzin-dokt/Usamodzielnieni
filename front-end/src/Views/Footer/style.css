.footer {
    background-color: #7345A9;
    color: white;
    height: 180px;
    position: absolute;
    right: 0;
    bottom: 0;
    left: 0;
    padding: 1rem;
    text-align: right;
    clip-path: polygon(98% 0%, 0% 80%, 0% 100%, 99% 100%);
}
.footer-icon {
    width: 40px;
    height: 40px;
    margin-right: 30px;
}

@media (max-width: 600px) {
    .footer {
        background-color: #7345A9;
        color: white;
        height: 140px;
        position: absolute;
        right: 0;
        bottom: 0;
        left: 0;
        padding: 1rem;
        text-align: right;
        clip-path: polygon(100% 0%, 0% 50%, 0% 100%, 100% 100%);
    }
    .footer-icon {
        width: 30px;
        height: 30px;
        margin-right: 10px;
    }
}