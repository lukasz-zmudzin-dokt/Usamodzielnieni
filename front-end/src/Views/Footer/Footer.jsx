import React from "react";
import {Col, Container, Row} from "react-bootstrap";
import "./style.css";

import kulczyk from "assets/footer/kulczyk-foundation-logo.png";
import pw from "assets/footer/politechnika-warszawska-logo.png";
import oneday from "assets/footer/oneday-logo.png";

import yt from "assets/footer/youtube-icon.png";
import tt from "assets/footer/twitter-icon.png";
import fb from "assets/footer/facebook-icon.png";
import ig from "assets/footer/instagram-icon.png";

const Footer = () => {
    return (
        <Container fluid className="font full-width">
            <Row className="p-5">
                <Col xs={{ span: 12, order: 2 }} sm={{ span: 12, order: 2 }} md={{ span: 3, order: 1}} className="p-4">
                    <Row className="company-title">
                        Partner
                    </Row>
                    <Row className="company-logo-row">
                        <img src={kulczyk} className="company-logo" alt="Kulczyk" />
                    </Row>
                </Col>
                <Col xs={{ span: 12, order: 3 }} sm={{ span: 12, order: 3 }} md={{ span: 3, order: 2}} className="p-4">
                    <Row className="company-title">
                        Twórca technologiczny <br/> strony studenci
                    </Row>
                    <Row className="company-logo-row">
                        <img src={pw} className="company-logo" alt="pw" />
                    </Row>
                </Col>
                <Col xs={{ span: 12, order: 4 }} sm={{ span: 12, order: 4 }} md={{ span: 3, order: 3}} className="p-4">
                    <Row className="company-title">
                        Pomysłodawca, <br/> twórca treści:
                    </Row>
                    <Row className="company-logo-row">
                        <img src={oneday} className="company-logo" alt="oneday" />
                    </Row>
                </Col>
                <Col xs={{ span: 12, order: 1 }} sm={{ span: 12, order: 1 }} md={{ span: 2, order: 4, offset: 1 }} className="social-media-col">
                    <Row className="pt-3">
                        <a href="https://www.youtube.com/channel/UCbKE3BfzmHGfXbB9nmmswYw" target="_blank" className="d-flex align-items-end">
                            <img src={yt} className="social-media-icon" alt="yt" />
                            youtube
                        </a>
                    </Row>
                    <Row className="pt-3">
                        <a href="https://twitter.com/usamodzielnieni" target="_blank" className="d-flex align-items-end">
                            <img src={tt} className="social-media-icon" alt="tt" />
                            twitter
                        </a>
                    </Row>
                    <Row className="pt-3">
                        <a href="https://www.facebook.com/Usamodzielnieni-110914023943342" target="_blank" className="d-flex align-items-end">
                            <img src={fb} className="social-media-icon lol" alt="fb"/>
                            facebook
                        </a>
                    </Row>
                    <Row className="pt-3">
                        <a href="https://www.instagram.com/_usamodzielnieni_" target="_blank" className="d-flex align-items-end">
                            <img src={ig} className="social-media-icon" alt="ig" />
                            instagram
                        </a>
                    </Row>
                </Col>
            </Row>
        </Container>
    );
};

export default Footer;