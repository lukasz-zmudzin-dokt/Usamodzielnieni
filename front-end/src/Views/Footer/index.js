import React from "react";
import Container from "react-bootstrap/Container";
import FbIcon from "../../assets/facebook-128.ico";
import LinkedInIcon from "../../assets/linkedin-128.ico";
import YouTubeIcon from "../../assets/youtube-128.ico";
import Background from "../../assets/menu-backdrop.png";

import "Views/Footer/style.css";

class Footer extends React.Component {

    render() {
        console.log(Background);
        return (
            <Container className="Footer">
                <div className="footer">
                    <div className="row justify-content-end mt-4">
                        <div className="col-8 mr-5">
                            <h2>KONTAKT</h2>
                        </div>
                    </div>
                    <div className="row justify-content-end mt-3">
                        <div className="col-8 align-self-end mr-5 pr-5">
                            <a href="https://www.facebook.com/fundacjaoneday/">
                                <img src={FbIcon} alt="" className="footer-icon"/>
                            </a>
                            <a href="https://www.linkedin.com/company/fundacja-one-day/about/">
                                <img src={LinkedInIcon} alt="" className="footer-icon"/>
                            </a>
                            <a href="https://www.youtube.com/channel/UCvWPmhiC5Q3gYVR3MPilHbQ">
                                <img src={YouTubeIcon} alt="" className="footer-icon"/>
                            </a>
                        </div>
                    </div>
                </div>
            </Container>
        );
    }
}

export default Footer;
