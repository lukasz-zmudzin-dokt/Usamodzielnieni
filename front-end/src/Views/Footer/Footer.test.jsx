import React from "react";
import { render, waitForElement } from "@testing-library/react";
import Footer from "./Footer";

describe("Footer", () => {
  it("should match snapshot", async () => {
    const { container, getByText } = render(<Footer />);
    await waitForElement(() => getByText("instagram"));
    expect(container).toMatchSnapshot();
  });
});
