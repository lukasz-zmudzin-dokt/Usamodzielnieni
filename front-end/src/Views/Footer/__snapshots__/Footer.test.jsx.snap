// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer should match snapshot 1`] = `
<div>
  <div
    class="font full-width container-fluid"
  >
    <div
      class="p-5 row"
    >
      <div
        class="p-4 col-md-3 col-sm-12 col-12 order-md-1 order-sm-2 order-2"
      >
        <div
          class="company-title row"
        >
          Partner
        </div>
        <div
          class="company-logo-row row"
        >
          <img
            alt="Kulczyk"
            class="company-logo"
            src="kulczyk-foundation-logo.png"
          />
        </div>
      </div>
      <div
        class="p-4 col-md-3 col-sm-12 col-12 order-md-2 order-sm-3 order-3"
      >
        <div
          class="company-title row"
        >
          Twórca technologiczny 
          <br />
           strony studenci
        </div>
        <div
          class="company-logo-row row"
        >
          <img
            alt="pw"
            class="company-logo"
            src="politechnika-warszawska-logo.png"
          />
        </div>
      </div>
      <div
        class="p-4 col-md-3 col-sm-12 col-12 order-md-3 order-sm-4 order-4"
      >
        <div
          class="company-title row"
        >
          Pomysłodawca, 
          <br />
           twórca treści:
        </div>
        <div
          class="company-logo-row row"
        >
          <img
            alt="oneday"
            class="company-logo"
            src="oneday-logo.png"
          />
        </div>
      </div>
      <div
        class="social-media-col col-md-2 col-sm-12 col-12 order-md-4 offset-md-1 order-sm-1 order-1"
      >
        <div
          class="pt-3 row"
        >
          <a
            class="d-flex align-items-end"
            href="https://www.youtube.com/channel/UCbKE3BfzmHGfXbB9nmmswYw"
            target="_blank"
          >
            <img
              alt="yt"
              class="social-media-icon"
              src="youtube-icon.png"
            />
            youtube
          </a>
        </div>
        <div
          class="pt-3 row"
        >
          <a
            class="d-flex align-items-end"
            href="https://twitter.com/usamodzielnieni"
            target="_blank"
          >
            <img
              alt="tt"
              class="social-media-icon"
              src="twitter-icon.png"
            />
            twitter
          </a>
        </div>
        <div
          class="pt-3 row"
        >
          <a
            class="d-flex align-items-end"
            href="https://www.facebook.com/Usamodzielnieni-110914023943342"
            target="_blank"
          >
            <img
              alt="fb"
              class="social-media-icon lol"
              src="facebook-icon.png"
            />
            facebook
          </a>
        </div>
        <div
          class="pt-3 row"
        >
          <a
            class="d-flex align-items-end"
            href="https://www.instagram.com/_usamodzielnieni_"
            target="_blank"
          >
            <img
              alt="ig"
              class="social-media-icon"
              src="instagram-icon.png"
            />
            instagram
          </a>
        </div>
      </div>
    </div>
  </div>
</div>
`;
